tests/js/panelsTests.js

```diff
@@ -411,104 +411,207 @@ https://github.com/gpii/universal/LICENSE.txt
+        gpii.tests.utils.verifyTooltipContents(panelName + " panel, choice input", that.locate("choiceInput"), that.model.choice, that.tooltip.model.idToContent, that.options.controlValues.choice, that.options.stringArrayIndex.choice, that.options.messageBase);
     };
 
-    gpii.tests.speakTextTester.verifyRendering = function (that) {
-        jqUnit.assertEquals("The instructions should have been set correctly.", that.options.messageBase.speakTextInstructions, that.locate("instructions").text());
-        gpii.tests.utils.verifyRadioButtonRendering(that.locate("choiceInput"), that.locate("choiceLabel"), that.options.choiceLabels, that.model.speakChoice);
-        gpii.tests.speakTextTester.verifyTooltip(that);
+    gpii.tests.yesNoTester.verifyRendering = function (that, panelName) {
+        var messages = that.options.messageBase;
+        var controlValues = that.options.controlValues.choice;
+        jqUnit.assertEquals(panelName + ": The instructions should have been set correctly.", messages.instructions, that.locate("instructions").text());
+        gpii.tests.utils.verifyRadioButtonRendering(that.locate("choiceInput"), that.locate("choiceLabel"), [messages[controlValues[0]], messages[controlValues[1]]], that.model.choice);
+        gpii.tests.yesNoTester.verifyTooltip(that, panelName);
     };
 
-    gpii.tests.speakTextTester.verifyModel = function (that, expectedValue) {
-        jqUnit.assertEquals("The model value should have been set correctly", expectedValue, that.model.speak);
+    gpii.tests.yesNoTester.verifyModel = function (that, panelName, expectedModel) {
+        fluid.each(expectedModel, function (value, path) {
+            jqUnit.assertEquals(panelName + ": The model value for path " + path + " has been set to " + value, value, fluid.get(that.model, path));
+        });
     };
 
+    /**************************
+     * Speak Text Panel Tests *
+     **************************/
+
+    fluid.defaults("gpii.tests.firstDiscovery.panel.speakText", {
+        gradeNames: ["gpii.firstDiscovery.panel.speakText", "autoInit"],
+        messageBase: {
+            "instructions": "Do you prefer to have items on the screen read out loud to you?",
+            "no": "No",
+            "yes": "Yes",
+            "yes-tooltip": "Select to turn voice on",
+            "no-tooltip": "Select to turn voice off",
+            "yes-tooltipAtSelect": "Voice is on",
+            "no-tooltipAtSelect": "Voice is off"
+        },
+        model: {
+            speak: true
+        }
+    });
+
+    fluid.defaults("gpii.tests.speakTextPanel", {
+        gradeNames: ["fluid.test.testEnvironment", "autoInit"],
+        components: {
+            yesNo: {
+                type: "gpii.tests.firstDiscovery.panel.speakText",
+                container: ".gpiic-fd-speakText"
+            },
+            yesNoTester: {
+                type: "gpii.tests.yesNoTester",
+                options: {
+                    expectedModels: {
```

Comment:
The expected models are the same, so you could probably reuse them by putting them as the default option of the base grade for the tester.